pysteps/blending/steps.py

```diff
@@ -633,37 +645,14 @@ def forecast(
         # 2.3.3 If zero_precip_radar, make sure that precip_cascade does not contain
         # only nans or infs. If so, fill it with the zero value.
         if zero_precip_radar:
-            precip_cascade = np.nan_to_num(
-                precip_cascade,
-                copy=True,
-                nan=np.nanmin(precip_models_cascade),
-                posinf=np.nanmin(precip_models_cascade),
-                neginf=np.nanmin(precip_models_cascade),
-            )
-
-        # 2.3.4 Check if the NWP fields contain nans or infinite numbers. If so,
-        # fill these with the minimum value present in precip (corresponding to
-        # zero rainfall in the radar observations)
-        (
-            precip_models_cascade,
-            precip_models_pm,
-            mu_models,
-            sigma_models,
-        ) = _fill_nans_infs_nwp_cascade(
-            precip_models_cascade,
-            precip_models_pm,
-            precip_cascade,
-            precip,
-            mu_models,
-            sigma_models,
-        )
+            precip_cascade[~np.isfinite(precip_cascade)] = np.nanmin(precip)
```

Comment:
```suggestion
            precip_cascade[~np.isfinite(precip_cascade)] = np.nanmin(precip_models)
```

However, I expect that the filler value should be the decomposed ``precip_models`` value as this is a normalized value which is used in the ``precip_cascade`` (also decomposed and normalized).

Reply:
Ah yes, this used to be `precip_models_cascade`, but that is not available anymore at this point in the code (if you supply just the regular model precipitation data. I am not quite sure what to do with this zero value business, as there is a zero value as well in the metadata, but that is rarely used as `np.nanmin(values)` is a good alternative.

In this case: what would be the downside of using `np.nanmin(precip_models)` here? Would it be easy to turn this into a normalized value?

Reply:
@mats-knmi, sorry for the slow reply here! I think the biggest difference is that ``precip_cascade`` is normalized to zero mean and standard deviation of one, so the min value may not be the same as in ``precip_models``. However, if I have read the code well, you could just use ``precip_cascade`` here, as it should always have a value somewhere in the grid, right? I'm not 100% sure about this statement, so good to double check, but if all values are infinites, we should be treating it as zero precipitation anyway.

Reply:
Hmmm yes, but the comment above this specifically says `make sure that precip_cascade does not contain only nans or infs`, which to me seems to indicate that it could only contain nans or infs... But actually I have been wondering for some time now why we would only fill nans or infs when there is no rain in the radar image. It would be sensible to fill nans and infs anyways right (if they are present)?

Reply:
I made a change  to check if `precip_cascade` is only nans/infs. If not we can use `nanmin(precip_cascade)`, but else we have to use `nanmin(precip_models_cascade)`, which means we have to look for a timestep (and model member) in the model field that has precipitation so that we can decompose that to get this value.

I am pretty sure this should always work and not hurt performance a lot. It may be a little bit overcomplicated though, so if we can do without that would still be nice.

Reply:
Looks good - this could work, I think. The alternative would be to save a zero value up front and just use that in the else statement. What do you think?

Reply:
That would be neater, but I don't know what this zero value would be, because a zero value for a cascade is different than for a regular precip array right?

Reply:
Yes, that's true indeed. Perhaps best to just leave it as you have proposed for now. :)